tornado_rest_client/utils.py

```diff
@@ -66,13 +68,14 @@ def populate_with_tokens(string, tokens, left_wrapper='%', right_wrapper='%',
     # First things first, swap out all instances of %<str>% with any matching
     # token variables found. If no items are in the hash (none, empty hash,
     # etc), then skip this.
-    allowed_types = (str, unicode, bool, int, float)
+    allowed_types = six.string_types + (six.text_type, str, bool, int, float)
+
     if tokens:
-        for k, v in tokens.iteritems():
+        for k, v in six.iteritems(tokens):
 
             if type(v) not in allowed_types:
-                log.warning('Token %s=%s is not in allowed types: %s' % (
-                    k, v, allowed_types))
+                log.warning('Token %s=%s (%s)is not in allowed types: %s' % (
+                    k, v, type(v), allowed_types))
                 continue
 
             string = string.replace(
```

Comment:
if k is `bytes` this will turn weird

```
In [1]: '%s%s%s' % ('hi', b'bye', 'hi')
Out[1]: "hib'bye'hi"```

Reply:
ditto for `v`

```
In [2]: str(b'bye')
Out[2]: "b'bye'"```

Reply:
i'd suggest only working with unicode.  `bytes` are for shipping things over the wire and should be immediately be turned into text (unicode) as soon as received.